src/frequenz/dispatch/_dispatcher.py

```diff
@@ -192,10 +195,32 @@ def __init__(
         self._empty_event = Event()
         self._empty_event.set()
 
+    @override
     def start(self) -> None:
         """Start the local dispatch service."""
         self._bg_service.start()
 
+    @property
+    @override
+    def is_running(self) -> bool:
+        """Whether the local dispatch service is running."""
+        return self._bg_service.is_running
+
+    @override
+    async def wait(self) -> None:
+        """Wait until all actor dispatches are stopped."""
+        await self._empty_event.wait()
+
+    @override
+    def cancel(self, msg: str | None = None) -> None:
+        """Stop the local dispatch service."""
+        self._bg_service.cancel(msg)
+
+        for instance in self._actor_dispatchers.values():
+            instance.cancel()
+
+        self._actor_dispatchers.clear()
```

Comment:
This should be done in the `await` instead, as `stop` is `.cancel()` + `await`, so if you do it like this, stop won't really wait until all background tasks are done.